aiida_nanotech_empa/workflows/cp2k/adsorbed_gw_ic_workchain.py

```diff
@@ -0,0 +1,285 @@
+from aiida_nanotech_empa.utils import common_utils
+
+import numpy as np
+
+from aiida.engine import WorkChain, ExitCode, calcfunction, ToContext, if_
+from aiida.orm import Int, Str, Bool, Code, List, Float, Dict
+from aiida.orm import StructureData
+
+from aiida.plugins import WorkflowFactory
+
+Cp2kMoleculeGwWorkChain = WorkflowFactory('nanotech_empa.cp2k.molecule_gw')
+Cp2kMoleculeOptWorkChain = WorkflowFactory('nanotech_empa.cp2k.molecule_opt')
+
+
+def _extract_image_plane(ase_geom):
```

Comment:
The whole concept is OK for purely physisorbed systems, this is teh target of course.
1)add mandatory check whether the system is physisorbed or not
2)add mandatory check whether the system is an allowed/known surface (for which we have image plane parameters) or not